src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1402,11 +1402,12 @@ private void ConvertMetadataReferencesToProjectReferences_NoLock(ProjectId proje
             {
                 if (CanConvertMetadataReferenceToProjectReference(projectIdToRetarget, referencedProjectId: projectId))
                 {
-                    foreach (PortableExecutableReference reference in modifiedSolution.GetProject(projectIdToRetarget).MetadataReferences)
+                    // PERF: call GetProjectState instead of GetProject, otherwise creating a new project might force all
+                    // Project instances to get created.
```

Comment:
i think Sam might have a PerfSensitive attribute he likes for cases like this.

Reply:
@sharwell In this case, this was one of the biggest hot bits we found looking at the performance of my rewrite, ironically. So this one popped in traces first.

Reply:
In that case, it would be good to document the test strategy and link the attribute.